### ISSUE TITLE
Rule proposal: `prefer-path2d`

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#1272]
Before [`Path2D`](https://developer.mozilla.org/en-US/docs/Web/API/Path2D/Path2D) was added, when doing animation with lots of `CanvasRenderingContext2D` draw methods, we have to put in a function and call it once and once again. But `Path2D` can be reused, it should be recommanded.

## Fail

```js
function draw() {
	context.moveTo(220, 60);
	context.arc(170, 60, 50, 0, 2 * Math.PI);
	// ...
	context.stroke();
}

function step() {
	draw();
	if (foo) {
		requestAnimationFrame(step);
	}
}

requestionAnimationFrame(step)
```

## Pass

```js
const path = new Path2D();
path.moveTo(220, 60);
path.arc(170, 60, 50, 0, 2 * Math.PI);

function step() {
	context.stroke(path);
	if (foo) {
		requestAnimationFrame(step);
	}
}

requestionAnimationFrame(step)
```

Not sure if it's doable in ESLint, maybe we can check a function called many `CanvasRenderingContext2D` draw method without variables, and suggest `Path2D`.

**Comment by sindresorhus** [comment]
This is now accepted.

**Comment by sindresorhus** [comment]
This rule would be so easy to do with type info, but I guess we can make some assumptions. Like `context` variable name, and certain method names like `.moveTo`, `.arc`, `.stroke`, ...
